### PR TITLE
[ARCTIC-915][AMS] Orphan files clean service should not delete flink temporary metadata files

### LINKED ISSUE #915: [Bug]: Flink can not restore from checkpoint if pending files are removed by Optimizer
### What happened?

Flink commiter would produce some pending manifest files. They would be committed(cleaned) after ckp completed.
If these files are removed by Optimizer because they are orphan files, flink task will not be able to recover from the checkpoint anymore.

### Affects Versions

master

### What engines are you seeing the problem on?

_No response_

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

fix #915 

## Brief change log

  - not delete flink temporary metadata files which start with `flink.job-id` in the latest snapshot summary
  - refactor `OrphanFilesCleanService`, remove the useless  execute/metadata/mode flag

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java [hunk 1]
```diff
@@ -107,107 +108,106 @@ public void run() {
...
         deleteFilesCnt += deleteInvalidDataFiles(table.io(),
             fileStatus,
             lastTime,
-            exclude,
-            execute);
+            exclude);
       }
     }
-    LOG.info("{} total delete[execute={}] {} files", table.id(), execute, deleteFilesCnt);
+    LOG.info("{} total delete {} files", table.id(), deleteFilesCnt);
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java:157]: Log seems cannot tell the table store information.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java [hunk 2]
```diff
@@ -107,107 +108,106 @@ public void run() {
...
-    LOG.info("{} total delete[execute={}] {} manifestList/manifest/metadata files", table.id(), execute,
+    LOG.info("{} total delete {} manifestList/manifest/metadata files", table.id(),
         deleteFilesCnt);
   }
 
+  private static Pattern getExcludeFileNameRegex(UnkeyedTable table) {
+    String latestFlinkJobId = null;
+    for (Snapshot snapshot : table.snapshots()) {
+      String flinkJobId = snapshot.summary().get("flink.job-id");
+      if (StringUtils.isNotBlank(flinkJobId)) {
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java:183]: I recommend using `com.google.common.base.Strings#isNullOrEmpty`.

Comment (wangtaohz) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java]: OK, I changed it to `!Strings.isNullOrEmpty(flinkJobId)`.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java (hunk 1 above)
Comment (wangtaohz) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java]: I modified some logs in the `OrphanFilesCleanService`.

However, I didn't find a convenient way to distinguish `ChangeTable`/`BaseTable` from `UnkeyedTable` by table's name or identifier.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**